src/UseCases/clients.js

```diff
@@ -0,0 +1,49 @@
+const Clients = require('../Models/clients');
+const bcrypt = require('../Lib/bcrypt');
+const jwt = require('../Lib/jwt');
+const sgMail = require('../Lib/sendgrid')
+
+function getClients(){
+   return Clients.find({})
+}
+
+function getById(idClient){
+   return Clients.findById(idClient)
+}
+
+ async function create(dataClients){
+   const {name, email, lastName} = dataClients
+   
+   const userFound = await getClients.findOne({email:email})
+   if (!userFound) throw new Error('Not permision to create, this client already exist') 
+   
+   const isValidMailUser = await Clients.compare(email, userFound.email)
+   if (!isValidMailUser) throw new Error('Not permision to create, this email already exist')
+   
+   const clientCreated = await Clients.create({...dataClients})
+
+   const token = jwt.sign({id: clientCreated._id})
+
+    sgMail.emailResetPassword({name, lastName, email, token})
```

Comment:
@KristomeisterDev Esto es un evento asíncrono por lo que hay que agregarle el await

Y hay que regresar algo.

```suggestion
    await sgMail.emailResetPassword({name, lastName, email, token})
    
    return "Correo enviado"
```